File: Examples/CSharp/ProgrammersGuide/Working-Diagrams/ExportPageToImage.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Diagram. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Diagram;
using Aspose.Diagram.Saving;

namespace CSharp.Diagrams
{
    public class ExportPageToImage
    {
        public static void Run()
        {
            //ExStart:ExportPageToImage
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Diagrams();

            Diagram diagram = new Diagram(dataDir + "ExportPageToImage.vsd");

            //Save diagram as PNG
            ImageSaveOptions options = new ImageSaveOptions(SaveFileFormat.PNG);

            // Save one page only, by page index
            options.PageIndex = 0;

            //Save resultant Image file
            diagram.Save(dataDir + "output.png", options);
            //ExEnd:ExportPageToImage
        }
    }
}

File: Examples/VisualBasic/Load-Save-Convert/ExportPageToImage.vb
Imports Microsoft.VisualBasic
Imports System.IO
Imports Aspose.Diagram
Imports Aspose.Diagram.Saving

Namespace Diagrams
    Public Class ExportPageToImage
        Public Shared Sub Run()
            ' ExStart:ExportPageToImage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_LoadSaveConvert()
            ' Load diagram
            Dim diagram As New Diagram(dataDir & "ExportPageToImage.vsd")

            ' Save diagram as PNG
            Dim options As New ImageSaveOptions(SaveFileFormat.PNG)

            ' Save one page only, by page index
            options.PageIndex = 0

            ' Save resultant Image file
            diagram.Save(dataDir & "ExportPageToImage_out.png", options)
            ' ExEnd:ExportPageToImage
        End Sub
    End Class
End Namespace